enterprise/admin/views.py

```diff
@@ -291,132 +292,396 @@ def _handle_bulk_upload(cls, enterprise_customer, manage_learners_form, request,
+        """
+        Create messages for the users who were not able to be enrolled in a course or program.
+
+        Args:
+            users: An iterable of users who were not successfully enrolled
+            enrolled_in (str): A string identifier for the course or program with which enrollment was attempted
+
+        Returns:
+            tuple: A 2-tuple containing a message type and message text
+        """
+        failed_emails = [user.email for user in users]
+        return (
+            'error',
+            _('Enrollment of some users in {enrolled_in} failed: {user_list}').format(
+                enrolled_in=enrolled_in,
+                user_list=', '.join(failed_emails),
+            )
+        )
+
+    @classmethod
+    def get_pending_message(cls, pending_users, enrolled_in):
+        """
+        Create messages for the users who were enrolled in a course or program.
+
+        Args:
+            users: An iterable of PendingEnterpriseCustomerUsers who were successfully linked with a pending enrollment
+            enrolled_in (str): A string identifier for the course or program the pending users were linked to
+
+        Returns:
+            tuple: A 2-tuple containing a message type and message text
+        """
+        pending_emails = [pending_user.user_email for pending_user in pending_users]
+        return (
+            'warning',
+            _(
+                'The following users do not have an account on {platform_name}. They have not been '
+                'enrolled in {enrolled_in}. When these users create an account, they will be '
+                'enrolled automatically: {pending_email_list}'
+            ).format(
+                platform_name=settings.PLATFORM_NAME,
+                enrolled_in=enrolled_in,
+                pending_email_list=', '.join(pending_emails),
+            )
+        )
+
+    @classmethod
+    def _enroll_users(
+            cls,
+            request,
+            enterprise_customer,
+            emails,
+            mode,
+            course_id=None,
+            program_details=None,
+            notify=True
+    ):
+        """
+        Enroll the users with the given email addresses to the courses specified, either specifically or by program.
+
+        Args:
+            cls (type): The EnterpriseCustomerManageLearnersView class itself
+            request: The HTTP request the enrollment is being created by
+            enterprise_customer: The instance of EnterpriseCustomer whose attached users we're enrolling
+            emails: An iterable of strings containing email addresses to enroll in a course
+            mode: The enrollment mode the users will be enrolled in the course with
+            course_id: The ID of the course in which we want to enroll
+            program_details: Details about a program in which we want to enroll
+            notify: Whether to notify (by email) the users that have been enrolled
```

Comment:
I think it does, for now. We built the notification setting so that other options besides email can be added later on; if we do that, then we'd want to make the docstring here more generic.